geoengine/ml.py

```diff
@@ -56,3 +67,64 @@ def register_ml_model(onnx_model: SerializableModel,
         model = MlModel(name=model_config.name, upload=str(upload_id), metadata=model_config.metadata,
                         display_name=model_config.display_name, description=model_config.description)
         ml_api.add_ml_model(model, _request_timeout=register_timeout)
+
+
+def validate_model_config(onnx_model: ModelProto, *,
+                          input_type: RasterDataType,
+                          output_type: RasterDataType,
+                          num_input_bands: int):
+    '''Validates the model config. Raises an exception if the model config is invalid'''
+
+    def check_data_type(data_type: TypeProto, expected_type: RasterDataType, prefix: 'str'):
+        if not data_type.tensor_type:
+            raise InputException('Only tensor input types are supported')
+        elem_type = data_type.tensor_type.elem_type
+        if elem_type != RASTER_TYPE_TO_ONNX_TYPE[expected_type]:
+            elem_type_str = tensor_dtype_to_string(elem_type)
+            raise InputException(f'Model {prefix} type `{elem_type_str}` does not match the '
+                                 f'expected type `{expected_type}`')
+
+    for domain in onnx_model.opset_import:
+        if domain.domain != '':
+            continue
+        if domain.version != 9:
+            raise InputException('Only ONNX models with opset version 9 are supported')
+
+    try:
+        model_ref = ReferenceEvaluator(onnx_model)
```

Comment:
I am not sure if this is the correct way of checking the model... zipMap works fine in our ONNX operator.

Reply:
https://github.com/onnx/sklearn-onnx/issues/218

ZipMap is debatable. Is it important?